app/pages/nics/Class4DivingExemptPage.scala

```diff
@@ -26,10 +26,17 @@ import queries.Settable
 case object Class4DivingExemptPage extends NicsBasePage[List[BusinessId]] {
   override def toString: String = "class4DivingExempt"
 
-  override def nextPageInNormalMode(userAnswers: UserAnswers, businessId: BusinessId, taxYear: TaxYear): Call =
-    if (Class4NonDivingExemptPage.remainingBusinesses(userAnswers).isEmpty)
-      routes.NICsCYAController.onPageLoad(taxYear)
-    else routes.Class4NonDivingExemptController.onPageLoad(taxYear, NormalMode)
+  override def nextPageInNormalMode(userAnswers: UserAnswers, businessId: BusinessId, taxYear: TaxYear): Call = {
```

Comment:
This is good. We need the similar logic on the CYA page so when a user wants to change their answer it knows to take them to the Class4NonDivingExemptController or the Class4NonDivingExemptSingleBusinessController